core/parse.h

```diff
@@ -31,67 +20,94 @@ struct parser_settings {
+
+	sqlite3 *sql_handle;			/* for SQL based parsers */
+	event_allocation_t event_allocation;
+};
+#define cur_event event_allocation.event
+
+void init_parser_state(struct parser_state *state);
+void free_parser_state(struct parser_state *state);
 
 /* the dive table holds the overall dive list; target table points at
  * the table we are currently filling */
 extern struct dive_table dive_table;
-extern struct dive_table *target_table;
-
-extern int metric;
 
 int trimspace(char *buffer);
 void start_match(const char *type, const char *name, char *buffer);
 void nonmatch(const char *type, const char *name, char *buffer);
-void event_start(void);
-void event_end(void);
-struct divecomputer *get_dc(void);
-
-bool is_dive(void);
-void reset_dc_info(struct divecomputer *dc);
-void reset_dc_settings(void);
-void settings_start(void);
-void settings_end(void);
-void dc_settings_start(void);
-void dc_settings_end(void);
-void dive_site_start(void);
-void dive_site_end(void);
-void dive_start(void);
-void dive_end(void);
-void trip_start(void);
-void trip_end(void);
-void picture_start(void);
-void picture_end(void);
-void cylinder_start(void);
-void cylinder_end(void);
-void ws_start(void);
-void ws_end(void);
-
-void sample_start(void);
-void sample_end(void);
-void divecomputer_start(void);
-void divecomputer_end(void);
-void userid_start(void);
-void userid_stop(void);
+void event_start(struct parser_state *state);
+void event_end(struct parser_state *state);
+struct divecomputer *get_dc(struct parser_state *state);
+
+bool is_dive(struct parser_state *state);
+void reset_dc_info(struct divecomputer *dc, struct parser_state *state);
+void reset_dc_settings(struct parser_state *state);
+void settings_start(struct parser_state *state);
+void settings_end(struct parser_state *state);
+void dc_settings_start(struct parser_state *state);
+void dc_settings_end(struct parser_state *state);
+void dive_site_start(struct parser_state *state);
+void dive_site_end(struct parser_state *state);
+void dive_start(struct parser_state *state);
+void dive_end(struct parser_state *state);
+void trip_start(struct parser_state *state);
+void trip_end(struct parser_state *state);
+void picture_start(struct parser_state *state);
+void picture_end(struct parser_state *state);
+void cylinder_start(struct parser_state *state);
+void cylinder_end(struct parser_state *state);
+void ws_start(struct parser_state *state);
+void ws_end(struct parser_state *state);
+
+void sample_start(struct parser_state *state);
+void sample_end(struct parser_state *state);
+void divecomputer_start(struct parser_state *state);
+void divecomputer_end(struct parser_state *state);
+void userid_start(struct parser_state *state);
+void userid_stop(struct parser_state *state);
 void utf8_string(char *buffer, void *_res);
 
-void add_dive_site(char *ds_name, struct dive *dive);
+void add_dive_site(char *ds_name, struct dive *dive, struct parser_state *state);
```

Comment:
+1 for moving suffix to state - e.g. dive_site_suffix.

Reply:
Stupid question: Is this `SSRF_WS` thing the webservice that was recently removed or is this something completely different? I.e. could we simply remove the code?

Reply:
Stupid question: Is this `SSRF_WS` thing the webservice that was recently removed or is this something completely different? I.e. could we simply remove the code?

Reply:
Stupid question: Is this `SSRF_WS` thing the webservice that was recently removed or is this something completely different? I.e. could we simply remove the code?

Reply:
Stupid question: Is this `SSRF_WS` thing the webservice that was recently removed or is this something completely different? I.e. could we simply remove the code?

Reply:
^ seems like so, but would defer to @dirkhh on the decision.
(github is spazzing today i see comments repeated 6 times)

Reply:
Stupid question: Is the whole `SSRF_WS` business related to the web-service that was removed recently? I.e. could the whole code simply be deleted?

Reply:
Argh.. Sorry. I tried again and again and it would not display anything for me. Perhaps my browser.

Reply:
no, GitHub was unwell - I saw the emails about you commenting, but couldn't see the comments on GitHub. Maybe some of their distributed infrastructure had outdated caches or something.
Anyway, yes, that SSRF_WS was indeed about the soon-to-be-discontinued GPS web service. I merged the commit removing it.